spec/server/functional/classrooms.spec.coffee

```diff
@@ -488,29 +488,34 @@ describe 'POST /db/classroom/:id/invite-members', ->
-    @teacher = yield utils.initUser()
-    @student1 = yield utils.initUser()
-    @student2 = yield utils.initUser()
-    @levelA = new Level({name: 'Level A', permissions: [{target: @artisan._id, access: 'owner'}]})
-    @levelA.set('original', @levelA._id)
-    @levelA = yield @levelA.save()
-    @levelB = new Level({name: 'Level B', permissions: [{target: @artisan._id, access: 'owner'}]})
-    @levelB.set('original', @levelB._id)
-    @levelB = yield @levelB.save()
-    @classroom = yield new Classroom({name: 'Classroom', ownerID: @teacher._id, members: [@student1._id, @student2._id] }).save()
-    @session1A = yield new LevelSession({creator: @student1.id, state: { complete: true }, level: {original: @levelA._id}, permissions: [{target: @student1._id, access: 'owner'}]}).save()
-    @session1B = yield new LevelSession({creator: @student1.id, state: { complete: false }, level: {original: @levelB._id}, permissions: [{target: @student1._id, access: 'owner'}]}).save()
-    @session2A = yield new LevelSession({creator: @student2.id, state: { complete: true }, level: {original: @levelA._id}, permissions: [{target: @student2._id, access: 'owner'}]}).save()
-    @session2B = yield new LevelSession({creator: @student2.id, state: { complete: false }, level: {original: @levelB._id}, permissions: [{target: @student2._id, access: 'owner'}]}).save()
+    yield utils.clearModels([CourseInstance, Course, User, Classroom, Campaign, Level, LevelSession])
+    @teacher = yield utils.initUser({role: 'teacher'})
+    admin = yield utils.initAdmin()
+    yield utils.loginUser(admin)
+    @levelA = yield utils.makeLevel({type: 'course'})
+    @levelB = yield utils.makeLevel({type: 'course'})
+    @campaignA = yield utils.makeCampaign({}, {levels: [@levelA]})
+    @campaignB = yield utils.makeCampaign({}, {levels: [@levelB]})
+    @courseA = yield utils.makeCourse({free: true, releasePhase: 'released'}, {campaign: @campaignA})
+    @courseB = yield utils.makeCourse({free: true, releasePhase: 'released'}, {campaign: @campaignB})
+    @student1 = yield utils.initUser({role: 'student'})
+    @student2 = yield utils.initUser({role: 'student'})
+    @session1A = yield new LevelSession({creator: @student1.id, state: { complete: true }, level: {original: @levelA._id.toHexString()}, permissions: [{target: @student1._id, access: 'owner'}]}).save()
+    @session1B = yield new LevelSession({creator: @student1.id, state: { complete: false }, level: {original: @levelB._id.toHexString()}, permissions: [{target: @student1._id, access: 'owner'}]}).save()
+    @session2A = yield new LevelSession({creator: @student2.id, state: { complete: true }, level: {original: @levelA._id.toHexString()}, permissions: [{target: @student2._id, access: 'owner'}]}).save()
+    @session2B = yield new LevelSession({creator: @student2.id, state: { complete: false }, level: {original: @levelB._id.toHexString()}, permissions: [{target: @student2._id, access: 'owner'}]}).save()
```

Comment:
Thanks for spreading the good word! However, the link points to client factories, which expect Backbone models. Server tests, which work with Mongoose and save to the test db, should use factories in [`/spec/server/utils.coffee`](https://github.com/codecombat/codecombat/blob/00d6d588ecbbef347de06fcfb8b524a4d99f8a6a/spec/server/utils.coffee#L103).